### ISSUE TITLE
Add parameters to allow allocating of specific executor ports

### ISSUE THREAD

**Issue by abhay-agarwal** [mesos/elasticsearch#185]
Currently the model is to pick two ports at random, but it would be nice to specify two specific ports instead of two ports in general. This may be some command line parameter passed to the scheduler.

**Comment by philwinder** [comment]
@abhay-agarwal Thank you for your suggestion. The reason why port allocation is random is to provide stability and resiliency. 
1. If another process is using that port, it will fail.
2. If an executor task dies for some reason, and tries to restart with your defined port, it might not be able to because the old instance is still holding onto that port.
3. Anything that is fixed (ip addresses or ports) is not scalable
4. Port allocation is handled by Mesos. The port that you specify may not be allowed due to Mesos rules.

Basically, by only allowing randomised ports, this forces the user to have a more resilient solution. Our overall aim is to provide users with a resilient system.

Port numbers can be accessed from the scheduler GUI, Mesos or the Zookeeper zNode.

**Comment by abhay-agarwal** [comment]
Hi Phil,

I understand the idea of using randomized ports completely. However, in order to provide for certain environments where ports must be clearly defined beforehand and be the same across the entire cluster (i.e. Amazon ELB, etc) this should be a feature (but clearly labeled with buyer beware as you point out). Furthermore, once IP-per-container hits, this will become much more valuable. Please see the kubernetes implementation of Elasticsearch for reference.
